### DISCUSSION TITLE
Catching breakpoints with netcoredbg running in docker container.

### DISCUSSION THREAD

**Discussion by leblocks** [mfussenegger/nvim-dap#1067] [Q&A]
I having some trouble finishing configuration of _nvim-dap_ for _netcoredbg_ running in docker container. I'm able to connect to it, but weren't able to catch any breakpoints. I have a feeling that the problem is in _pathMappings_ configuration.

So here is my setup, I've tried doing it both in windows and linux environments:

* project under debug - simple dotnet 6 project, created with
```sh
dotnet new console --name testapp
 ```

* Content of _Program.cs_:
```cs
using System.Threading.Tasks;

namespace Test;

public class Program - simple program that counts till 9999, just for debug purposes.
{
    public static async Task Main(string[] args)
    {
        int counter = 0;

        while (counter < 9999)
        {
            await Task.Delay(1000);
            System.Console.WriteLine($"Printing some number: {counter}");
            counter++;
        }
    }
}
```

* _Dockerfile_ - based on microsoft image, with installation of _netcoredbg_. Application root is mounted to _/opt/app_ folder.
```dockerfile
FROM mcr.microsoft.com/dotnet/sdk:6.0

# install debugger
RUN mkdir /opt/debugger \
    && cd /opt/debugger \
    && curl -LO https://github.com/Samsung/netcoredbg/releases/latest/download/netcoredbg-linux-amd64.tar.gz \
    && tar -xvf ./netcoredbg-linux-amd64.tar.gz \
    && rm ./netcoredbg-linux-amd64.tar.gz

WORKDIR /opt/app

CMD ["dotnet", "run"]
```
* _nvim-dap_ configuration - so I running _netcoredbg_ manually inside container with exposed port 4987 and connecting to it from neovim.
```lua
-- creating new adapter
dap.adapters.coreclrserver = {
    type = 'server',
    host = '127.0.0.1',
    port = 4987,
}

-- configuration
dap.configurations.cs = {
  {
    type = "coreclrserver";
    request = "attach";
    name = "Attach to process - netcoredbg (in docker)";
    pathMappings = {
        { 
            localRoot = vim.fn.getcwd();
            remoteRoot = '/opt/app';
        },
    };
    processId = function()
        return vim.fn.input('Process id ')
    end;
  }
}
```
* _netcoredbg_ launch:
```bash
#! /bin/bash -x

/opt/debugger/netcoredbg/netcoredbg \
    --interpreter=vscode \
    --server=4987
```

It seems that with my setup I'm able to connect to netcoredbg instance and even pause main thread if I want, but breakpoints aren't being hit for some reason. I assume that the problem is with _pathMappings_ setup.

Thanks.

**Comment by KiLLeRRaT** [comment]
Have you managed to get this sorted out?  I'm about to endeavour onto this mission myself... No point if it's never going to work I guess heh.

**Reply by leblocks** [reply]
I didn't touch it since then, If I manage to do it I'll write it here.
